src/constructors/createGlobalStyle.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import React from 'react'
+import PropTypes from 'prop-types'
+import { CONTEXT_KEY, STATIC_EXECUTION_CONTEXT } from '../constants'
+import _GlobalStyle from '../models/GlobalStyle'
+import StyleSheet from '../models/StyleSheet'
+import ServerStyleSheet from '../models/ServerStyleSheet'
+import type { CSSConstructor, Interpolation, Stringifier } from '../types'
+import withTheme from '../hoc/withTheme'
+import hashStr from '../vendor/glamor/hash'
+
+export default (stringifyRules: Stringifier, css: CSSConstructor) => {
+  const GlobalStyle = _GlobalStyle(stringifyRules)
+
+  const createGlobalStyle = (
+    strings: Array<string>,
+    ...interpolations: Array<Interpolation>
+  ) => {
+    const rules = css(strings, ...interpolations)
+    const id = `sc-global-${hashStr(JSON.stringify(rules))}`
+    const style = new GlobalStyle(rules, id)
+
+    class GlobalStyleComponent extends React.Component {
+      static contextTypes = {
+        [CONTEXT_KEY]: PropTypes.oneOfType([
+          PropTypes.instanceOf(StyleSheet),
+          PropTypes.instanceOf(ServerStyleSheet),
+        ]),
+      }
+
+      static styledComponentId = id
+
+      componentWillMount() {
+        const { props } = this
+        const sheet = this.context[CONTEXT_KEY] || StyleSheet.master
+        const context = style.isStatic ? STATIC_EXECUTION_CONTEXT : { ...props }
+
+        style.createStyles(context, sheet)
+      }
+
+      componentWillReceiveProps(props: { [prop: string]: any }) {
+        const sheet = this.context[CONTEXT_KEY] || StyleSheet.master
+        const context = style.isStatic ? STATIC_EXECUTION_CONTEXT : { ...props }
+
+        style.updateStyles(context, sheet)
+      }
+
+      componentWillUnmount() {
+        const sheet = this.context[CONTEXT_KEY] || StyleSheet.master
+        style.removeStyles(sheet)
+      }
+
+      render() {
+        return null
+      }
+    }
+
+    return withTheme(GlobalStyleComponent)
```

Comment:
There's currently an issue with using `createGlobalStyle` in production builds, where if the created component is not eventually a children of a `ThemeProvider`, you will get an error on [this](https://github.com/marionebl/styled-components/blob/3e2b5e812a1538cbed2365606103377ff3870fba/src/hoc/withTheme.js#L58) line, saying `subscribe` is undefined. It was very difficult to figure out, but it basically prevents us from running a production build using this new feature.

Here's before (not working):
```js
ReactDOM.render(
  <React.Fragment>
    <Normalize />
    <LockBody />
    <ThemeProvider theme={theme}>
      <ApolloProvider client={apolloClient}>
        <Router>
          <App />
        </Router>
      </ApolloProvider>
    </ThemeProvider>
  </React.Fragment>,
  ensure(document.getElementById('root')),
)
```

And after (working):
```js
ReactDOM.render(
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <Normalize />
      <LockBody />
      <ApolloProvider client={apolloClient}>
        <Router>
          <App />
        </Router>
      </ApolloProvider>
    </React.Fragment>
  </ThemeProvider>,
  ensure(document.getElementById('root')),
)
```

I would gladly contribute to the PR with a fix, but unfortunately I'm not comfortable enough in the codebase to know where to make this fix. If someone is willing to help me, I'm happy to contribute, though.

Reply:
Note, this bug only happens when building with `process.env.NODE_ENV` set to `production`.

Reply:
Adressed this via a64f2272d6d3d41e32baa0c747bfdcb6ffca1158
Released via `@marionebl/styled-components@3.3.0`
I also updated the demo at https://codesandbox.io/s/30m80qjwy1 to show this in action